### DISCUSSION TITLE
Thor and VFP Advanced x64

### DISCUSSION THREAD

**Discussion by HaPe4** [VFPX/Thor#254] [Polls]
Thor works great with VFP9 and VFP10 - Advanced from Chen.
Unfortunately, the Thor update process does not run with VFP10 x64 because the 32-bit FLL VFPCompression is used to unpack the archives. Is there a solution to this?
-- 
Hans-Peter Grözinger

**Comment by DougHennig** [comment]
1. Can you not just do the Thor update using the 32-bit version?
2. The latest version of Thor uses other unpacking mechanisms (Window Shell or PowerShell) if VFPCompression fails.

**Reply by Jimrnelson** [reply]
@DougHennig 

It actually tries it in a different order:

1. Shell.Application
2. PowerShell
3. VFPCompression.FLL

**Reply by HaPe4** [reply]
1. That's how I've been doing it so far. However, since there are different configurations in the different VFP directories, I would like to refrain from doing this. Something could be overwritten during synchronization.
2. I saw this in the source code. However, I don't trust myself to adjust it yet.

**Reply by Jimrnelson** [reply]
@HaPe4 

You might consider modifying thor_proc_extractfilesfromzip.prg and saving your modified version in the "My Tools" folder.

Thor will use your version of that prg instead of its own and will never modify it.

If you get it to work, you could then attach it here.

**Reply by HaPe4** [reply]
@Jimrnelson: Then I don't understand why it doesn't work. This may be because the SET LIBRARY TO VFPCompression.fll ist set, before checking whether it can be unpacked using the Windows Shell or PowerShell.

**Reply by HaPe4** [reply]
@Jimrnelson: Thanks, i think i am able to do this :-)

**Reply by HaPe4** [reply]
Once I've implemented it, I'll append the modified thor_proc_extractfilesfromzip.prg to the thread.

**Comment by lscheffler** [comment]
deleted

**Reply by Jimrnelson** [reply]
Attaching the prg to this thread is adequate.

**Reply by HaPe4** [reply]
Version of Thor_Proc_ExtractFilesFromZip.PRG from 27.11.2023 is able to unpack with VFPA64.
[Thor_Proc_ExtractFilesFromZip.zip](https://github.com/VFPX/Thor/files/14298469/Thor_Proc_ExtractFilesFromZip.zip)
Unfortunately, an error occurs due to a invalid format in the version file.

![Invalid_Format](https://github.com/VFPX/Thor/assets/29651009/f64bd782-9be8-4d13-bc2b-479ae688d640)

**Reply by HaPe4** [reply]
Sorry that I can't do much myself right now as I'm working on an urgent project.
I didn't expect your support to come so quickly.
This is really a great community.

**Reply by Jimrnelson** [reply]
@HaPe4 

Something funny going on here, this is a head scratcher.  Here are some thoughts when you find time to return to this:

1. The file you uploaded is in fact the current file.
2. The error image starts with this text "Invalid format in version file" - I cannot find this text in the current version of Thor, nor does it look familiar.

Thus, it seems likely to me that something has gone awry with your installation of Thor.

I suggest you download Thor.App directly (see [https://github.com/VFPX/Thor/blob/master/Docs/Thor_install.md](url)) and follow the first two steps (unzipping into your current Thor folder).
